boolean/boolean.py

```diff
@@ -1041,7 +1041,7 @@ def demorgan(self):
         This is achieved by canceling double NOTs and using De Morgan laws.
         """
         expr = self.cancel()
-        if expr.isliteral or not isinstance(expr.args[0], (self.NOT, self.AND, self.OR)):
+        if expr.isliteral or not isinstance(expr, self.NOT):
```

Comment:
It is a tad hard for me to understand what this small but important change does in other cases. Do you think you could add a few extra tests that match the changes you made here?

Reply:
ping?

Reply:
Sorry for taking so long to answer. I hope to add the tests you asked until tomorrow.

Reply:
That's very gentle of you! Boolean logic is hard to process at times for me ;)